src/tla/__rewire_finite_sets_ext_in_apalache.tla

```diff
@@ -0,0 +1,166 @@
+------------------------ MODULE FiniteSetsExt ---------------------------------
+\*------ MODULE __rewire_finite_sets_ext_in_apalache -----------------------
+(**
+ * ^^^^^^^^^^^^^^^^^^^^^^ We have to call this module FiniteSetsExt in any
+ * case, otherwise, SANY complains.
+ *
+ * This file contains alternative definitions for the operators defined in
+ * FiniteSetsExt. Most importantly, we are adding type annotations. We also
+ * define the Apalache-compatible behavior.
+ *
+ * These definitions are automatically rewired by the Apalache importer.
+ *
+ * Compare with the original definitions in FiniteSetsExt.tla:
+ *
+ * https://github.com/tlaplus/CommunityModules/blob/master/modules/FiniteSetsExt.tla
+ *)
+
+LOCAL INSTANCE Integers
+LOCAL INSTANCE FiniteSets
+LOCAL INSTANCE __apalache_folds 
+LOCAL INSTANCE Functions
+
+(**
+ * Fold op over the elements of set using base as starting value.
+ * Note that this FoldSet is different from Apalache!FoldSet.
+ *
+ * Example:
+ *         FoldSet(LAMBA x,y : x + y, 0, 0 .. 10) = 55
+ *
+ * @type: ((a, b) => b, b, Set(a)) => b;
+ *)
+FoldSet(op(_, _), base, set) ==
+    \* ApalacheFoldSet is accumulating the result in the left argument,
+    \* whereas FinitSetsExt!FoldSet is accumulating the result
+    \* in the right argument.
+    LET OpSwap(x, y) == op(y, x) IN
+    __ApalacheFoldSet(OpSwap, base, set)
+
+(**
+ * Calculate the sum of the elements in set.
+ *
+ * Example: Sum(0 .. 10) = 55
+ *
+ * @type: Set(Int) => Int;
+ *)
+SumSet(set) ==
+   LET Plus(x, y) == x + y IN
+   __ApalacheFoldSet(Plus, 0, set)
+
+(**
+ * Calculuate the product of the elements in set.
+ *
+ * Example: Product(1 .. 3) = 6
+ *
+ * @type: Set(Int) => Int;
+ *)
+ProductSet(set) ==
+   LET Mult(x, y) == x * y IN
+   __ApalacheFoldSet(Mult, 1, set)
+
+(**
+ * An alias for FoldSet. ReduceSet was used instead of FoldSet in
+ * earlier versions of the community modules.
+ *
+ * @type: ((a, b) => b, Set(a), b) => b;
+ *)
+ReduceSet(op(_, _), set, acc) == 
+   FoldSet(op, set, acc)
```

Comment:
Why are some operators using `FoldSet` and some `__ApalacheFoldSet` ?

Reply:
`FoldSet` is the operator defined right in this module

Reply:
See, this is the kind of confusion that happens when your spec language doesn't have namespaces.